script/fs-bench/image/run.sh

```diff
@@ -0,0 +1,186 @@
+  export GOPROXY="https://goproxy.cn,direct"
+  which process || GO111MODULE=on go build -o "/usr/local/bin/process" "./process/main.go"
+  which scrape || GO111MODULE=on go build -o "/usr/local/bin/scrape" "./scrape/main.go"
+  unset GOPROXY
+  popd
+  apt-get install -y --fix-missing gnuplot
+}
+buildTools
+
+echo "Building test image (2/2)..."
+buildTestImage() {
+  curl -s fs-bench-registry:5000/v2/fio/tags/list | grep -q stargz && return
+  ctr-remote image optimize --plain-http --entrypoint '["fio"]' \
+    --args '["--filename=/work/testfile","--size=512M","--rw=read","--name=test"]' \
+    "http://$IMAGE_LEGACY" "http://$IMAGE_STARGZ"
+}
+buildTestImage
+
+testrun() {
+    local image="${1}"
+    local config="${2}"
+    local output="${3}"
+    local pull="${4:-}"
+
+    local configDir=$(dirname "$config")
+    local configFile=$(basename "$config")
+    local tmpdir=$(mktemp -d)
+    local containerID=$(basename $(mktemp))
+
+    if [ -n "$pull" ] ; then
+        ctr-remote image "$pull" --plain-http "$image"
+    fi
+
+    mkdir -p "$output"
+    ctr-remote run --rm --mount type=bind,src=$configDir,dst=/config,options=rbind:ro \
+        --mount type=bind,src=$tmpdir,dst=/output,options=rbind:rw \
+        "$image" "$containerID" \
+        fio "/config/$configFile" >"$output/fio.log"
+    mv $tmpdir/* "$output"
+}
+
+set_noprefetch() {
+    local NOPREFETCH="${1}"
+    sed -i 's/noprefetch = .*/noprefetch = '"${NOPREFETCH}"'/g' "${REMOTE_SNAPSHOTTER_CONFIG_DIR}config.stargz.toml"
+}
+
+if [ ! -e /output/legacy ]; then
+  echo "Benchmarking legacy..."
+  mkdir -p /output/legacy
+  "${REBOOT_CONTAINERD_SCRIPT}" -nosnapshotter /output/legacy/metrics
+  testrun "$IMAGE_LEGACY" "$FIO_CONFIG_DIR/randread-1.conf" /output/legacy/randread-1 pull
+  testrun "$IMAGE_LEGACY" "$FIO_CONFIG_DIR/randread-4.conf" /output/legacy/randread-4
+  testrun "$IMAGE_LEGACY" "$FIO_CONFIG_DIR/seqread-1.conf" /output/legacy/seqread-1
+  testrun "$IMAGE_LEGACY" "$FIO_CONFIG_DIR/seqread-4.conf" /output/legacy/seqread-4
+fi
+
+echo "Limit bandwidth of eth0 to 10mbps"
+#tc qdisc del dev eth0 root
+tc qdisc add dev eth0 root tbf rate 10Mbps burst 100k latency 10ms
+
+if [ ! -e /output/stargz ]; then
+  echo "Benchmarking stargz..."
+  mkdir -p /output/stargz
+  set_noprefetch "true" # disable prefetch
+
+  "${REBOOT_CONTAINERD_SCRIPT}" "" /output/stargz/randread-1/metrics
+  testrun "$IMAGE_STARGZ" "$FIO_CONFIG_DIR/randread-1.conf" /output/stargz/randread-1 rpull
+
+  "${REBOOT_CONTAINERD_SCRIPT}" "" /output/stargz/randread-4/metrics
+  testrun "$IMAGE_STARGZ" "$FIO_CONFIG_DIR/randread-4.conf" /output/stargz/randread-4 rpull
+
+  "${REBOOT_CONTAINERD_SCRIPT}" "" /output/stargz/seqread-1/metrics
+  testrun "$IMAGE_STARGZ" "$FIO_CONFIG_DIR/seqread-1.conf" /output/stargz/seqread-1 rpull
+
+  "${REBOOT_CONTAINERD_SCRIPT}" "" /output/stargz/seqread-4/metrics
+  testrun "$IMAGE_STARGZ" "$FIO_CONFIG_DIR/seqread-4.conf" /output/stargz/seqread-4 rpull
+
+fi
```

Comment:
We should use real stargz image here. Stargz image can be converted from legacy one, using `--stargz-only` option for `ctr-remote image optimize` command.

Reply:
OK, I'll fix it.